lib/ddtrace/contrib/httpclient/instrumentation.rb

```diff
@@ -23,17 +23,14 @@ module InstanceMethods
           def do_get_block(req, proxy, conn, &block)
             host = req.header.request_uri.host
             request_options = datadog_configuration(host)
-            pin = datadog_pin(request_options)
-
-            return super unless pin
+            client_config = Datadog::Tracing.configuration_for(self)
 
             Datadog::Tracing.trace(Ext::SPAN_REQUEST, on_error: method(:annotate_span_with_error!)) do |span, trace|
               begin
-                request_options[:service_name] = pin.service_name
-                span.service = service_name(host, request_options)
+                span.service = service_name(host, request_options, client_config)
                 span.span_type = Datadog::Ext::HTTP::TYPE_OUTBOUND
 
-                if Datadog::Tracing.enabled? && !should_skip_distributed_tracing?(pin)
+                if Datadog::Tracing.enabled? && !should_skip_distributed_tracing?(client_config)
```

Comment:
This call to `Datadog::Contrib::HTTP.should_skip_distributed_tracing?` will raise an error if `client_config` is `nil`, which would happen if `client_config = Datadog::Tracing.configuration_for(self)` returns `nil` above.